lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,298 @@
+	name string
+
+	// amount is the amount in the Carol's wallet.
+	amount btcutil.Amount
+
+	// pushAmt is the amount to be pushed to Dave.
+	pushAmt btcutil.Amount
+
+	// feeRate is an optional fee in satoshi/bytes used when opening a
+	// channel.
+	feeRate btcutil.Amount
+
+	// carolBalance is Carol's expected balance in her channel before
+	// subtracting the reserved amount to pay for a commitment transaction.
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+
+	// commitmentType allows to define the exact type when opening the
+	// channel.
+	commitmentType lnrpc.CommitmentType
+
+	// private denotes if the channel opening is announced to the network or
+	// not.
+	private bool
+}
+
+// testChannelFundMax checks various channel funding scenarios where the user
+// instructed the wallet to use all remaining funds.
+func testChannelFundMax(net *lntest.NetworkHarness,
+	ht *harnessTest) {
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	args := nodeArgsForCommitType(lnrpc.CommitmentType_ANCHORS)
+	carol := net.NewNode(ht.t, "Carol", args)
+	defer shutdownAndAssert(net, ht, carol)
+
+	dave := net.NewNode(ht.t, "Dave", args)
+	defer shutdownAndAssert(net, ht, dave)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	net.EnsureConnected(ht.t, carol, dave)
+
+	// Calculate reserve amount for one channel.
+	ctxb := context.Background()
+	reserveResp, _ := carol.WalletKitClient.RequiredReserve(
+		ctxb, &walletrpc.RequiredReserveRequest{
+			AdditionalPublicChannels: 1,
+		},
+	)
+	reserveAmount := btcutil.Amount(reserveResp.RequiredReserve)
+
+	var testCases = []*chanFundMaxTestCase{
+		{
+			name:               "wallet amount is dust",
+			amount:             2_000,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount < min chan size (~18000sat)",
+			amount: 18_000,
+			// Using a feeRate of 1 sat/vByte ensures that we test
+			// for min chan size and not excessive fees.
+			feeRate:            1,
+			chanOpenShouldFail: true,
+		},
+
+		{
+			name:   "wallet amount > min chan size (37000sat)",
+			amount: 37_000,
+			// The transaction fee to open the channel must be
+			// subtracted from Carol's balance.
+			// (since wallet balance < max-chan-size)
+			carolBalance: btcutil.Amount(37_000) -
+				fundingFee(1, false),
+		},
+
+		{
+			name: "wallet amount > max chan size " +
+				"(20000000sat)",
+			amount:       20_000_000,
+			carolBalance: lnd.MaxFundingAmount,
+		},
+
+		// Checks that to push corresponding amount to remote the local
+		// amount is not increased above MaxFundingAmount.
+		{
+			name: "wallet amount > max chan size, " +
+				"push amount == max-chan-size",
+			amount:             20_000_000,
+			pushAmt:            lnd.MaxFundingAmount,
+			chanOpenShouldFail: true,
+		},
+
+		// Checks that enough local funds are kept in reserve before
```

Comment:
same here this testcase does not check for any reserve, which reserve is meant here in detail, this fails for the same reason as the testcase above in the function `NewChannelReservation`. To test reservation the channeltype should be Anchor. 

All in all I would suggest the testcases hunt for a specific error rather then an error in general, otherwise the reviewer has to dig into the tests and actually verify why this testcase is failing in the first place

Reply:
I will remove this test as it tests for the same error condition as the one before.